test/models/script_test.rb

```diff
@@ -192,12 +207,72 @@ class ScriptTest < ActiveSupport::TestCase
 
     script.destroy!
 
-    assert_difference 'Script.count' do
-      Script.import path
-    end
+    old_count = Script.count
+
+    ret = Script.import path
+
+    assert_equal old_count + 1, Script.count
+    assert_equal 1, ret.count
+    assert ret.first.valid?
+    assert Script.find_by uuid: uuid
+
+    FileUtils.rm path
+  end
+
+  test 'import a new script and existing script' do
+    old_count = Script.count
+
+    uuid   = SecureRandom.uuid
+    script = @script.dup
+
+    script.name = 'Should be imported as new'
+    script.uuid = uuid
+    script.save!
+
+    path = Script.where(id: [@script.id, script.id]).export
+
+    script.destroy!
+
+    @script.parameters.clear
+    @script.requires.clear
+    @script.update! name: 'Updated'
+
+    ret = Script.import path
 
+    assert_equal old_count + 1, Script.count
+    assert_equal 2, ret.count
+    assert ret.all? &:valid?
     assert Script.find_by uuid: uuid
 
+    assert_not_equal 'Updated', @script.reload.name
+    assert @script.parameters.any?
+    assert @script.requires.any?
+
+    FileUtils.rm path
+  end
+
+  test 'should not import an invalid script' do
+    old_count = Script.count
+
+    uuid = SecureRandom.uuid
+
+    invalid_script = @script.dup
+    invalid_script.name = 'valid name'
+    invalid_script.uuid = uuid
+    invalid_script.save!
+
+    invalid_script.update_attribute('name', '')
+
+    path = Script.where(id: invalid_script.id).export
+
+    invalid_script.destroy!
+
+    ret = Script.import path
```

Comment:
Y acá lo mismo, en lugar de guardar el `old_count` utilizaría algo como:
```suggestion
    assert_no_difference 'Script.count' do
      scripts = Script.import path
    end
```